nutrition-app/src/main/java/com/nutrition/controller/NutritionController.java

```diff
@@ -1,22 +1,81 @@
 package com.nutrition.controller;
 
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Controller;
 import org.springframework.ui.ModelMap;
+import org.springframework.validation.BindingResult;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.ModelAttribute;
 import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.servlet.ModelAndView;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.nutrition.model.NutritionModel;
+import com.nutrition.service.NutritionService;
+import java.util.*;
 
 @Controller
+@ControllerAdvice
 public class NutritionController {
+	@Autowired
+	NutritionService nutritionService;
+	NutritionModel fooddao;
 
-    @RequestMapping(value = "/hello_world", method = RequestMethod.GET)
-    public String printHelloWorld(ModelMap modelMap){
+	@RequestMapping("home")
+	public String home() {
+		return "home";
+	}
 
-        // add attribute to load modelMap
-        modelMap.addAttribute("message",
-                "Hello World and Welcome to Spring MVC!");
+	// ajax call
+	@GetMapping("foodnamelist")
+	@ResponseBody
+	public String getsearchfood(@RequestParam("foodName") String foodname) {
+		List<String> foodnamelist = new ArrayList<String>();
+		List<NutritionModel> list = nutritionService.getFoodList(foodname);
+		for (NutritionModel nutritionModel : list) {
+			foodnamelist.add(nutritionModel.getFoodname());
+		}
 
-        // return the name of the file to be loaded "hello_world.jsp"
-        return "hello_world";
-    }
+		ObjectMapper Obj = new ObjectMapper();
+		String jsonStr = null;
+		try {
+			jsonStr = Obj.writeValueAsString(foodnamelist);
+		} catch (JsonProcessingException e) {
+			// TODO Auto-generated catch block
+			e.printStackTrace();
+		}
```

Comment:
all of this should happen in the service layer. controller methods shouldn't hold any business logic.